src/Main.java

```diff
@@ -1,14 +1,250 @@
+                    double numberOf2 = (change1/apoen2);
+                    int numberOf2Int = (int)floor(numberOf2);
+                    if (numberOf2Int <= cashState2) {
+                        numApoen2 = numberOf2Int;
+
+                    }
+                    else {
+                        numApoen2 = cashState2;
+                    }
+                    change1 = change1 - (numApoen2 * apoen2);
+
+                }
+
+                if ((change1 / apoen1) >= 1) {
+                    double numberOf1 = (change1/apoen1);
+                    int numberOf1Int = (int)floor(numberOf1);
+                    if (numberOf1Int <= cashState1) {
+                        numApoen1 = numberOf1Int;
+                    }
+                    else {
+                        numApoen1 = cashState1;
+                    }
+                    change1 = change1 - (numApoen1 * apoen1);
+                }
+
+                if ((change1 / apoen05) >= 1) {
+                    double numberOf05 = (change1/apoen05);
+                    int numberOf05Int = (int)floor(numberOf05);
+                    if (numberOf05 <= cashState05) {
+                        numApoen05 = numberOf05Int;
+                    }
+                    else {
+                        numApoen05 = cashState05;
+                    }
+                    change1 = change1 - (numApoen05 * apoen05);
+                }
+
+                if ((change1 / apoen02) >= 1) {
+                    double numberOf02 = (change1/apoen02);
+                    int numberOf02Int = (int)floor(numberOf02);
+                    if (numberOf02 <= cashState02) {
+                        numApoen02 = numberOf02Int;
+                    }
+                    else {
+                        numApoen02 = cashState02;
+                    }
+                    change1 = change1 - (numApoen02 * apoen02);
+                }
 
+                if ((change1 / apoen01) >= 1) {
+                    double numberOf01 = (change1/apoen01);
+                    int numberOf01Int = (int)floor(numberOf01);
+                    if (numberOf01Int <= cashState01) {
+                        numApoen01 = numberOf01Int;
+                    }
+                    else {
+                        numApoen01 = cashState01;
+                    }
+                    change1 = change1 - (numApoen01 * apoen01);
+                }
 
+                if ((change1 / apoen001) >= 1) {
+                    double numberOf001 = (change1/apoen001);
+                    int numberOf001Int = (int)floor(numberOf001);
+                    if (numberOf001 <= cashState001) {
+                        numApoen001 = numberOf001Int;
+                    }
+                    else {
+                        numApoen001 = cashState001;
+                    }
+                    change1 = change1 - (numApoen001 * apoen001);
+                }
 
+                if ((change == change1) || (change1>0)) {
```

Comment:
change 1 is a variable which is not clear what it contains, and why is it different then change